packages/grafana-data/src/types/datasource.ts

```diff
@@ -412,7 +413,7 @@ export interface QueryEditorHelpProps<TQuery extends DataQuery = DataQuery> {
  */
 export type LegacyResponseData = TimeSeries | TableData | any;
 
-export type DataQueryResponseData = DataFrame | DataFrameDTO | LegacyResponseData;
+export type DataQueryResponseData = DataFrame | DataFrameDTO | LegacyResponseData | StreamingResponseData;
```

Comment:
this change makes me nervous -- as it becomes the valid response for all datasources!

In general, the fewer changes we can do in `grafana-data` the better

Reply:
Do you think it then makes sense to change the interface of `GrafanaLiveSrv` and drop the generic `DataQueryResponse` https://github.com/grafana/grafana/blob/c780854a18100fa9df64dd8b9609452e04f3f72c/packages/grafana-runtime/src/services/live.ts#L46 in favour of a new `StreamingDataQueryResponse` contained in `live` package?

Reply:
changed as discussed